include/kframework/lemmas/lemmas.k

```diff
@@ -92,6 +93,63 @@ module LEMMAS [symbolic]
  // Hardcoded #addrFromPrivateKey simplifications, see: https://github.com/runtimeverification/haskell-backend/issues/3573
     rule #addrFromPrivateKey("0x0000000000000000000000000000000000000000000000000000000000000001") => 721457446580647751014191829380889690493307935711 [priority(40)]
 
+  // ########################
+  // Memory
+  // ########################
+
+    rule 0 <=Int #memoryUsageUpdate(MU, START, WIDTH) => true requires 0 <=Int MU andBool 0 <=Int START andBool 0 <=Int WIDTH [simplification]
+
+    rule #memoryUsageUpdate(#memoryUsageUpdate(MU, START1, WIDTH1), START2, WIDTH2) => #memoryUsageUpdate(MU, START1, WIDTH1)
+      requires #rangeUInt(256, MU) andBool 0 <Int WIDTH1 andBool 0 <Int WIDTH2
+       andBool START2 +Int WIDTH2 <=Int START1 +Int WIDTH1
+      [simplification]
+
+    rule #memoryUsageUpdate(#memoryUsageUpdate(MU, START1, WIDTH1), START2, WIDTH2) => #memoryUsageUpdate(MU, START2, WIDTH2)
+      requires #rangeUInt(256, MU) andBool 0 <Int WIDTH1 andBool 0 <Int WIDTH2
+       andBool START1 +Int WIDTH1 <Int START2 +Int WIDTH2
+      [simplification]
+
+  // ########################
+  // Gas Costs
+  // ########################
+
+    rule 0 <=Int Csload(_, _)                   => true                                       [simplification]
+    rule 0 <=Int Cmem(_, N)                     => true requires 0 <=Int N                    [simplification]
+    rule 0 <=Int Caddraccess(_, _)              => true                                       [simplification]
+    rule 0 <=Int Csstore(_, _, _, _)            => true                                       [simplification]
+    rule 0 <=Int Cgascap(_, _:Int, _:Int, _)    => true                                       [simplification]
+    rule 0 <=Gas Cgascap(_, _, _, _)            => true                                       [simplification]
+    rule 0 <=Int #allBut64th(_:Int)             => true                                       [simplification]
+    rule 0 <=Gas #allBut64th(_)                 => true                                       [simplification]
```

Comment:
```suggestion
    rule 0 <=Int Cgascap(_, _:Int, _:Int, _)    => true                                       [simplification]
    rule 0 <=Int #allBut64th(_:Int)             => true                                       [simplification]
```

Are these rules actually needed? It seems to me that instead what should happen is the `#gas(...)` on the right bubbles up to the top, then we have already have a rule that says:

```
rule _ <=Gas #gas(_) => true
```

Reply:
It still seems to me that the rules here with `<=Gas` are not needed.